container-chains/templates/frontier/runtime/src/xcm_config.rs

```diff
@@ -244,3 +258,228 @@ impl cumulus_pallet_dmp_queue::Config for Runtime {
     type XcmExecutor = XcmExecutor<XcmConfig>;
     type ExecuteOverweightOrigin = EnsureRoot<AccountId>;
 }
+
+parameter_types! {
+    // we just reuse the same deposits
+    pub const ForeignAssetsAssetDeposit: Balance = 0;
+    pub const ForeignAssetsAssetAccountDeposit: Balance = 0;
+    pub const ForeignAssetsApprovalDeposit: Balance = 0;
+    pub const ForeignAssetsAssetsStringLimit: u32 = 50;
+    pub const ForeignAssetsMetadataDepositBase: Balance = 0;
+    pub const ForeignAssetsMetadataDepositPerByte: Balance = 0;
+    pub CheckingAccount: AccountId = PolkadotXcm::check_account();
+}
+
+#[cfg(feature = "runtime-benchmarks")]
+/// Simple conversion of `u32` into an `AssetId` for use in benchmarking.
+pub struct ForeignAssetBenchmarkHelper;
+#[cfg(feature = "runtime-benchmarks")]
+impl pallet_assets::BenchmarkHelper<AssetId> for ForeignAssetBenchmarkHelper {
+    fn create_asset_id_parameter(id: u32) -> AssetId {
+        id.try_into()
+            .expect("number too large to create benchmarks")
+    }
+}
+#[cfg(feature = "runtime-benchmarks")]
+impl pallet_asset_rate::AssetKindFactory<AssetId> for ForeignAssetBenchmarkHelper {
+    fn create_asset_kind(id: u32) -> AssetId {
+        id.try_into()
+            .expect("number too large to create benchmarks")
+    }
+}
+
+pub type AssetId = u16;
```

Comment:
Is there a specific reason why we use u16 here instead of u128 as we do in Moonbeam?

Reply:
I think it might offer better protection to evm, so I am going to change it to u128

Reply:
Actually it might provide more protection if we set it to u16, as it is more difficult for an attacker to guess an evm contract that has more FFs fixed

Reply:
For context:
- What worries me the most is that someone deploys a contract in the precompile address. However with a u16 (2 bytes) they need to deploy a contract at (0xFFFF..FF (18 bytes)assetId). That is 2^144 probability already.

Reply:
What do you mean by FFs?